### PR TITLE
Load graffiti from file

### PR DESCRIPTION
**Pull request by terencechain**
Accompany design doc: https://hackmd.io/9su3BOB8TvOMITOsqHtO_A?view

Upon approval I will add a section for this in: https://docs.prylabs.network/docs/prysm-usage

Tested all the possible scenarios in run time with interop mode

### CODE REVIEW

File: validator/client/propose.go [hunk 1]
```diff
@@ -257,3 +267,47 @@ func signVoluntaryExit(
...
+		r.Seed(time.Now().Unix())
+		i := r.Uint64() % uint64(len(v.graffitiStruct.Random))
+		return []byte(v.graffitiStruct.Random[i]), nil
+	}
+
+	// Last and when specified, individual validator specified graffiti will be used.
+	idx, err := v.validatorClient.ValidatorIndex(ctx, &ethpb.ValidatorIndexRequest{PublicKey: pubKey[:]})
+	if err != nil {
+		return []byte{}, err
+	}
```
Comment (prestonvanloon) [inline validator/client/propose.go:306]: Can this be cached some how? We know this information from the duties as well

File: validator/node/node.go [hunk 2]
```diff
@@ -363,6 +371,17 @@ func (s *ValidatorClient) registerClientService(
...
 	}
+
+	gStruct := &g.Graffiti{}
+	var err error
+	if s.cliCtx.IsSet(flags.GraffitiFileFlag.Name) {
+		n := s.cliCtx.String(flags.GraffitiFileFlag.Name)
+		gStruct, err = g.ParseGraffitiFile(n)
+		if err != nil {
+			return err
+		}
```
Comment (prestonvanloon) [inline validator/node/node.go:382]: Maybe just log a warning here? It would be sad if parsing a graffiti file failed and the validator remained offline

Comment (terencechain) [reply validator/node/node.go:382]: Good point!

File: validator/client/propose.go (hunk 1 above)
Comment (terencechain) [reply validator/client/propose.go]: Discussed offline, there's no cache but this is ok. The call is cheap and proposing block is not a frequent event

File: validator/graffiti/parse_graffiti.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+)
+
+type Graffiti struct {
+	Default  string            `yaml:"default,omitempty"`
+	Random   []string          `yaml:"random,omitempty"`
+	Specific map[uint64]string `yaml:"specific,omitempty"`
```
Comment (prestonvanloon) [inline validator/graffiti/parse_graffiti.go:12]: In your design, you call this "validators". Why is it "specific" here? I

Comment (terencechain) [reply validator/graffiti/parse_graffiti.go:12]: I changed it to specific in the design doc but i forgot to update the example sections.

You mentioned specific in our chat and I liked it better than validators

Comment (prestonvanloon) [reply validator/graffiti/parse_graffiti.go:12]: Sounds good. No preference here, just wanting consistency. thanks!

**Review by prestonvanloon — APPROVED** [review]

File: endtoend/helpers/helpers.go [hunk 4]
```diff
@@ -83,6 +85,20 @@ func WaitForTextInFile(file *os.File, text string) error {
 	}
 }
 
+func GraffitiYamlFile(testDir string) (string, error) {
+	b := []byte(`default: "Rice"
+random: 
+  - "Sushi"
+  - "Ramen"
+  - "Takoyaki"
+`)
```
Comment (prestonvanloon) [inline endtoend/helpers/helpers.go:94]: yum

**Pull request merged**